app/src/renderer/components/staking/PageValidator.vue

```diff
@@ -250,6 +296,50 @@ export default {
         }
       }
     },
+    async submitUndelegation({ amount }) {
+      const candidateId = this.validator.owner
+
+      const currentlyDelegated = calculateTokens(
+        this.validator,
+        this.delegation.committedDelegates[candidateId] || 0
+      )
+
+      const delegation = [
+        {
+          atoms: currentlyDelegated.minus(amount),
+          delegate: this.validator
+        }
+      ]
+      console.log(delegation)
+
+      try {
+        await this.$store.dispatch("submitDelegation", delegation)
```

Comment:
I did a refactor on the `submitDelegation` function on the other PR to simplify it. We should sync to integrate that here